src/main/scala/chisel3/util/experimental/BoringUtils.scala

```diff
@@ -256,6 +257,17 @@ object BoringUtils {
     def drill(source: A, path: Seq[BaseModule], connectionLocation: Seq[BaseModule], up: Boolean): A = {
       path.zip(connectionLocation).foldLeft(source) {
         case (rhs, (module, conLoc)) if (module.isFullyClosed) => boringError(module); DontCare.asInstanceOf[A]
+        case (rhs: Element, (module, _))
+            if ((up || isDriveDone(reifySingleData(rhs).get)) && module == path(0) && isPort(
+              reifySingleData(rhs).get
+            ) &&
+              (!createProbe.nonEmpty || !createProbe.get.writable)) => {
+          // So far we can handle ports that are views of single elements only.
+          reifySingleData(rhs).get.asInstanceOf[A]
```

Comment:
```suggestion
        case (rhs: Element, (module, _))
            if ((up || isDriveDone(reifySingleData(rhs).get)) && module == path(0) && isPort(
              reify(rhs)
            ) &&
              (!createProbe.nonEmpty || !createProbe.get.writable)) => {
          // So far we can handle ports that are views of single elements only.
          reify(rhs).asInstanceOf[A]
```
Since `rhs` is an `Element` you should be able to use `reify` which is `Element => Element`.